src/Design.cc

```diff
@@ -190,24 +190,104 @@ float Design::getNetCap(odb::dbNet* net, sta::Corner* corner, MinMax minmax)
   return pin_cap + wire_cap;
 }
 
-bool Design::isSequential(odb::dbMaster* master)
+sta::LibertyCell* Design::getLibertyCell(odb::dbMaster* master)
 {
   sta::dbSta* sta = getSta();
   sta::dbNetwork* network = sta->getDbNetwork();
 
   sta::Cell* cell = network->dbToSta(master);
   if (!cell) {
+    return nullptr;
+  }
+  return network->libertyCell(cell);
+}
+
+bool Design::isBuffer(odb::dbMaster* master)
+{
+  auto lib_cell = getLibertyCell(master);
+  if (!lib_cell) {
     return false;
   }
+  return lib_cell->isBuffer();
+}
 
-  sta::LibertyCell* lib_cell = network->libertyCell(cell);
+bool Design::isInverter(odb::dbMaster* master)
+{
+  auto lib_cell = getLibertyCell(master);
   if (!lib_cell) {
     return false;
   }
+  return lib_cell->isInverter();
+}
 
+bool Design::isSequential(odb::dbMaster* master)
+{
+  auto lib_cell = getLibertyCell(master);
+  if (!lib_cell) {
+    return false;
+  }
   return lib_cell->hasSequentials();
 }
 
+float Design::staticPower(odb::dbInst* inst, sta::Corner* corner)
+{
+  sta::dbSta* sta = getSta();
+  sta::dbNetwork* network = sta->getDbNetwork();
+
+  sta::Instance* sta_inst = network->dbToSta(inst);
+  if (!sta_inst) {
+    return 0.0;
+  }
+  // sta::LibertyCell* lib_cell = network->libertyCell(sta_inst);
+  sta::PowerResult power = sta->power(sta_inst, corner);
+  return power.leakage();
+}
+
+float Design::dynamicPower(odb::dbInst* inst, sta::Corner* corner)
+{
+  sta::dbSta* sta = getSta();
+  sta::dbNetwork* network = sta->getDbNetwork();
+
+  sta::Instance* sta_inst = network->dbToSta(inst);
+  if (!sta_inst) {
+    return 0.0;
+  }
+  // sta::LibertyCell* lib_cell = network->libertyCell(sta_inst);
+  sta::PowerResult power = sta->power(sta_inst, corner);
+  return (power.internal() + power.switching());
+}
+
+bool Design::isInClock(odb::dbInst* inst)
```

Comment:
Do you intend for FF to return true based on the clock pin?

Reply:
We will include FF as well. So it will return true. As long as is_seq is true we should be good CircuitOps can differentiate them.